src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -1403,7 +1406,11 @@ def create_service_principal_for_rbac(
 
     graph_client = _graph_client_factory(cmd.cli_ctx)
     role_client = _auth_client_factory(cmd.cli_ctx).role_assignments
-    scopes = scopes or ['/subscriptions/' + role_client.config.subscription_id]
+
+    if role and not scopes:
+        logger.warning(SCOPE_WARNING)
+        scopes = ['/subscriptions/' + role_client.config.subscription_id]
```

Comment:
Unlike https://github.com/Azure/azure-cli/pull/20924, the default value assignment is done in our own function, instead of letting knack assign the default value and set `is_default`.